util/osshare/filesharingstatus_windows.go

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2021 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// +build windows
+
+package osshare
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"golang.org/x/sys/windows/registry"
+)
+
+const (
+	sendFileShellKey = `*\shell\tailscale`
+)
+
+var (
```

Comment:
this is fine, but another style I often use is: 

var ipnExePath struct {
       sync.Mutex
       cache string // populated lazily on first use
}

then:

```go
   ipnExePath.Lock()
   defer ipnExePath.Unlock()

   if ipnExePath.cache != nil {
       return ipnExePath.cache, nil
   }

    // outdent the rest of the code rather than nesting a bunch ....
```